flocker/node/agents/ebs.py

```diff
@@ -682,13 +786,38 @@ def _next_device(self, instance_id, volumes, devices_in_use):
 
     def create_volume(self, dataset_id, size):
         """
+        Create a volume on EBS backend.
+        """
+        return self.create_volume_with_profile(
+            dataset_id, size, MandatoryProfiles.DEFAULT.value)
+
+    def create_volume_with_profile(self, dataset_id, size, profile_name):
+        """
         Create a volume on EBS. Store Flocker-specific
         {metadata version, cluster id, dataset id} for the volume
         as volume tag data.
         Open issues: https://clusterhq.atlassian.net/browse/FLOC-1792
         """
-        requested_volume = self.connection.create_volume(
-            size=int(Byte(size).to_GiB().value), zone=self.zone)
+        requested_size = int(Byte(size).to_GiB().value)
+        try:
+            volume_type, iops = _volume_type_and_iops_for_profile_name(
+                profile_name, requested_size)
+            requested_volume = self.connection.create_volume(
+                size=requested_size,
+                zone=self.zone,
+                volume_type=volume_type,
+                iops=iops)
+        except EC2ResponseError as e:
+            CREATE_VOLUME_FAILURE(dataset_id=dataset_id,
+                                  aws_code=e.code,
+                                  aws_message=e.message).write()
+            volume_type, iops = _volume_type_and_iops_for_profile_name(
```

Comment:
Reading tests suggests this is here to cover cases where size doesn't match profile requirements. I don't think we should be retrying in other cases. Possibly it's better to not retry at all, and just document this restriction for now? Would users prefer outright failure or would they prefer wrong profile? I suspect the former but maybe I'm wrong.